tdrs-backend/tdpservice/parsers/aggregates.py

```diff
@@ -55,3 +55,27 @@ def case_aggregates_by_month(df, dfs_status):
     aggregate_data['rejected'] = ParserError.objects.filter(file=df).filter(case_number=None).count()
 
     return aggregate_data
+
+
+def total_errors_by_month(df, dfs_status):
+    """Return total errors for each month in the reporting period."""
+    calendar_year, calendar_qtr = fiscal_to_calendar(df.year, df.quarter)
+    month_list = transform_to_months(calendar_qtr)
+
+    total_errors_data = {"months": []}
+
+    for month in month_list:
+        if dfs_status == "Rejected":
+            total_errors_data["months"].append(
+                {"month": month, "total_errors": "N/A"})
+            continue
+
+        month_int = month_to_int(month)
+        rpt_month_year = int(f"{calendar_year}{month_int}")
+
+        error_count = ParserError.objects.filter(
```

Comment:
we should be able to query all ParserErrors before running the for loop, and have the counts in a table.
 
Another note: as we are parsing more and more files, this query would take forever to run, the problem is df is indexed easily but the rpt_month_year will be slow.

Reply:
i pulled the main `ParserError.objects.all()` out of the loop - are you suggesting grouping by rpt_month_year or something to that effect as well?